src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/BasicHtmlWebResponseObject.Common.cs

```diff
@@ -14,16 +14,27 @@ namespace Microsoft.PowerShell.Commands
     /// <summary>
     /// Response object for html content without DOM parsing.
     /// </summary>
-    public class BasicHtmlWebResponseObject : WebResponseObject
+    public partial class BasicHtmlWebResponseObject : WebResponseObject
     {
         #region Private Fields
 
-        private static Regex s_attribNameValueRegex;
-        private static Regex s_attribsRegex;
-        private static Regex s_imageRegex;
-        private static Regex s_inputFieldRegex;
-        private static Regex s_linkRegex;
-        private static Regex s_tagRegex;
+        [GeneratedRegex(@"([^""'>/=\s\p{Cc}]+)(?:\s*=\s*(?:""(.*?)""|'(.*?)'|([^'"">\s]+)))?", RegexOptions.Singleline | RegexOptions.IgnoreCase)]
+        private static partial Regex s_attribNameValueRegex();
```

Comment:
I don't see the point of this change. The original Regex is compiled and works just as fast, but it is more compact.

Reply:
And what are the benefits? The final code is still noticeably larger, but the performance does not increase.

Reply:
I will revert this change

Reply:
There's **significant** difference between `RegexOptions.Compiled` and `GeneratedRegex`! More about them here https://devblogs.microsoft.com/dotnet/regular-expression-improvements-in-dotnet-7/. They will not only reduce startup time but also increase performance.

Reply:
> I can't simply put it into words how much better generated regex is,

When it comes to code performance neither words nor emotions work - you have to prove with tests that a particular code will become smaller and faster.

In this particular case, the code will only get bigger because PowerShell is dynamic and can never completely get rid of the Regex compiler. The initial improvements won't be noticeable since the initialization of the cmdlet takes a significant amount of time.

Where we would like Regex SG and where it is justified is the pwsh startup scenario. And there are indeed places where this SG can be applied.

Reply:
@iSazonov if you pointed me to such cases I could try to apply Regex SG

Reply:
> When it comes to code performance neither words nor emotions work - you have to prove with tests that a particular code will become smaller and faster.

You're correct here. I should have provided data for my claims.

> In this particular case, the code will only get bigger because PowerShell is dynamic and can never completely get rid of the Regex compiler. The initial improvements won't be noticeable since the initialization of the cmdlet takes a significant amount of time.
> 
> Where we would like Regex SG and where it is justified is the pwsh startup scenario. And there are indeed places where this SG can be applied.

I didn't realize this code wasn't used in the pwsh itself, I admit my mistake. In this case delaying parsing and doing all the complex IL emitting is justified, if this code is not called in a cmdlet. 

This also makes me want to compare the size of the IL emitted for `RegexOptions.Compiled` and the IL compiled by Roslyn for the source generated one.. interesting.